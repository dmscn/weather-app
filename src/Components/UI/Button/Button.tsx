import React, { Component, ReactNode } from 'react';
import classNames from 'classnames';
import Button from '@material-ui/core/Button';

import styles from './Button.module.css';

export default class UIButton extends Component<any> {
  render() {
    const { className = "", children, ...rest } = this.props;
    return (
      <div className={classNames(styles.Container, this.props.className)}>
        <Button
          color="primary"
          variant="contained"
          {...rest}
        >
          {children}
        </Button>
      </div>
    )
  }
}
