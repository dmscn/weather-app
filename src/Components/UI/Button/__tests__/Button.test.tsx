import React from 'react'
import { shallow } from 'enzyme';

import Button from '../Button';


describe('<Button />', () => {
  it('renders', () => {
    const wrapper = shallow(<Button>Button</Button>);
    expect(wrapper).toMatchSnapshot();
  });
});