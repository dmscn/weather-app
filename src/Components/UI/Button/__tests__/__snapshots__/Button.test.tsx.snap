// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UIButton>
    Button
  </UIButton>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Button)
        color="primary"
        variant="contained"
      >
        Button
      </WithStyles(Button)>,
      "className": "Container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Button",
        "color": "primary",
        "variant": "contained",
      },
      "ref": null,
      "rendered": "Button",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Button)
          color="primary"
          variant="contained"
        >
          Button
        </WithStyles(Button)>,
        "className": "Container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Button",
          "color": "primary",
          "variant": "contained",
        },
        "ref": null,
        "rendered": "Button",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
